Wikipedia/UI-V5/WMFSearchViewController.m

```diff
@@ -75,17 +75,32 @@ - (void)updateSearchStateAndNotifyDelegate:(WMFSearchState)state {
     }
 
     self.state = state;
-    [self updateRecentSearchesVisibility];
+    [self updateRecentSearchesVisibility:YES];
 
     [self.delegate searchController:self searchStateDidChange:self.state];
 }
 
 - (void)updateRecentSearchesVisibility {
-    if ([self.searchBar.text length] == 0 && self.recentSearchesViewController.recentSearchesItemCount > 0) {
-        [self.recentSearchesContainerView setHidden:NO];
-    } else {
-        [self.recentSearchesContainerView setHidden:YES];
+    [self updateRecentSearchesVisibility:YES];
+}
+
+- (void)updateRecentSearchesVisibility:(BOOL)animated {
+    BOOL hideRecentSearches =
+        [self.searchBar.text length] > 0 || self.recentSearchesViewController.recentSearchesItemCount == 0;
+
+    /*
+       HAX: Need to show/hide superviews since recent & results are in the same container. should use UIViewController
+          containment/transition API instead in the future.
+     */
+    if ((self.recentSearchesViewController.view.superview.alpha == 0) == hideRecentSearches) {
+        return;
     }
+
+    [UIView animateWithDuration:animated ? [CATransaction animationDuration] : 0.0
```

Comment:
`[CATransaction animationDuration]` is non-specific in a case where we should probably be specific and specify the timing we want.